PersonaStats.Rmd

```diff
@@ -0,0 +1,287 @@
+  as.data.frame() %>%
+  mutate(percentage_responses = round(Freq/nrow(resultsTidy)*100, 0))
+numSelected
+```
+
+## What were the most common single responses
+
+```{r}
+pivot_longer(kindOfWorkDF[which(kindOfWorkDF$num_descriptions == 1),], 
+             4:ncol(kindOfWorkDF)-2, 
+             names_to = "colnamesWere", 
+             values_to = "valuesWere") %>% 
+  select(valuesWere) %>% 
+  table() %>% 
+  data.frame() %>% 
+  mutate(percentage_singleDescription_responses = 
+           round(Freq/numSelected[which(numSelected$Var1 == 1), "Freq"]*100,0)) %>% 
+  arrange(-Freq)
+```
+
+## What were the most common pairs of responses (no matter how many kinds of work descriptions were provided)
+
+Look at [this solution from stackoverflow to find pairs](https://stackoverflow.com/a/62374554), but it only works to find pairs that are right next to each other, ugh. So just wrote my own function.
+
+note that kindOfWorkDF's first column is a timestamp, we don't want that one for this, and the last two columns are a count of number of NAs and number of descriptions given. We don't want those in the pairs either. 
+
+note also combn will also fail if there aren't enough non-NA values, so we want to only select rows with more than one description
+
+Given those two notes, pass this function the following input data frame
+
+`kindOfWorkDF[which(kindOfWorkDf$num_descriptions > 1),-c(1,(ncol(kindOfWorkDF)-1), ncol(kindOfWorkDF))]`
+
+```{r}
+find_pairs <- function(inputDF, rowOI ){
+  pairs <- c()
+  notNA_cols_combos <- t(combn(which(!is.na(inputDF[rowOI,])), 2)) #rows will be indices for each pair
+  for (rowIndex in 1:nrow(notNA_cols_combos)){
+    pairs <- c(pairs, paste0(unlist(inputDF[rowOI, notNA_cols_combos[rowIndex,]]), collapse="-"))
+  }
+  return(pairs)
+}
+```
+
+```{r}
+numSelected %<>% 
+  mutate(Var1 = as.integer(Var1))
+
+kindOfWorkDF_subset <- kindOfWorkDF[which(kindOfWorkDF$num_descriptions > 1),-c(1,(ncol(kindOfWorkDF)-1), ncol(kindOfWorkDF))]
+
+pairs_all <- unlist(lapply(1:nrow(kindOfWorkDF_subset), 
+                           function(x) find_pairs(kindOfWorkDF_subset, x))) %>%
+  as.data.frame() %>%
+  `colnames<-`(c("pairs")) %>%
+  group_by(pairs) %>%
+  mutate(count = n()) %>%
+  ungroup() %>%
+  distinct() %>%
+  arrange(-count) %>%
+  mutate(percentage_notSingleDescription_responses = 
+           round(count/sum(numSelected[which(numSelected$Var1 > 1), "Freq"])*100,0)
+         )
+print(pairs_all, n=nrow(pairs_all))
+  
+```
+
+This percentage sums to greater than 100, but I think because the pairs are made whether there a respondent provided 2, 3, or 5 descriptions so there will be more pairs than responses that had pairs possible, so not sure if the percentage is helpful or not. 
+
+## How many times was each description selected/reported
+
+find a vector with the count of number of responses for each description, add the descriptions as a column, turning this into a dataframe
+
+```{r}
```

Comment:
formatting looks a bit funky here..

Reply:
Yeah, I've tried the rstudio reformatting too and I've never gotten it to format a way that looks clean